v2/spanner-to-sourcedb/terraform/samples/spanner-to-sharded-mysql/README.md

```diff
@@ -0,0 +1,508 @@
+## Sample Scenario: Spanner to Sharded MySQL reverse replication
+
+> **_SCENARIO:_** This Terraform example illustrates launching a reverse replication
+> jobs to replicate spanner writes for a sharded MySQL source, setting up all the required cloud infrastructure.
+> **Only the source details of physical shards are needed as input.**
+
+## Terraform permissions
+
+In order to create the resources in this sample,
+the`Service account`/`User account` being used to run Terraform
+should have the
+required [permissions](https://cloud.google.com/iam/docs/manage-access-service-accounts#multiple-roles-console).
+There are two ways to add permissions -
+
+1. Adding pre-defined roles to the service account running Terraform.
+2. Creating a custom role with the granular permissions and attaching it to the
+   service account running Terraform.
+
+### Using custom role and granular permissions (recommended)
+
+Following permissions are required -
+
+```shell
+- compute.firewalls.create
+- compute.firewalls.delete
+- compute.firewalls.update
+- dataflow.jobs.cancel
+- dataflow.jobs.create
+- dataflow.jobs.updateContents
+- iam.roles.get
+- iam.serviceAccounts.actAs
+- pubsub.subscriptions.create
+- pubsub.subscriptions.delete
+- pubsub.topics.attachSubscription
+- pubsub.topics.create
```

Comment:
Are we still recommending pubsub based pipeline? The GCS based pipeline is what we should be recommending for per-shard ordering.